File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/dialog/notebookselection/NotebookSelectionPresenter.java
package com.github.android.lvrn.lvrnproject.view.dialog.notebookselection;

import androidx.recyclerview.widget.RecyclerView;

import com.github.valhallalabs.laverna.persistent.entity.Notebook;

import java.util.List;

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

public interface NotebookSelectionPresenter {

    void bindView(NotebookSelectionDialogFragment notebookSelectionDialogFragment);

    void unbindView();

    void subscribeRecyclerViewForPagination(RecyclerView recyclerView);

    void disposePagination();

    List<Notebook> getNotebooksForAdapter();
}

File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/dialog/notebookselection/NotebookSelectionPresenter.kt
package com.github.android.lvrn.lvrnproject.view.dialog.notebookselection

import androidx.recyclerview.widget.RecyclerView
import com.github.valhallalabs.laverna.persistent.entity.Notebook

/**
 * @author Vadim Boitsov <vadimboitsov1></vadimboitsov1>@gmail.com>
 */
interface NotebookSelectionPresenter {
    fun bindView(notebookSelectionDialogFragment: NotebookSelectionDialogFragment?)

    fun unbindView()

    fun subscribeRecyclerViewForPagination(recyclerView: RecyclerView?)

    fun disposePagination()

    val notebooksForAdapter: List<Notebook>
}
